### PR TITLE
Add github actions CI

### PR DESCRIPTION
**Pull request by hershen**
Add CI functionality via github actions.
Perform 3 steps:
1) Configure project with CMAKE.
2) Build project with CMAKE.
3) Run tests with ctest.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: hersh-CI
+on: [push, pull_request]
```
Comment (oorijsonar) [inline .github/workflows/ci.yml:2]: Will this run when a commit is pushed to an open PR?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  build_and_test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: configure
+        run: cmake -S . -B build
+      - name: build
+        run: cmake --build build -j2
+      - name: test
+        run: ctest --test-dir build
```
Comment (oorijsonar) [inline .github/workflows/ci.yml]: I think you need `--output-on-failure` otherwise you won't see what went wrong.

**Review by oorijsonar — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (hershen) [reply .github/workflows/ci.yml:2]: Yes.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (hershen) [reply .github/workflows/ci.yml]: Fixed.

**Review by oorijsonar — APPROVED** [review]

**Comment by oorijsonar** [comment]
You should monitor how many minutes this is taking up as you could get charged for using their runner I think.

**Comment by hershen** [comment]
It looks like the spending limit is $0 if you don't specify a payment method.

**Pull request merged**